zedUpload/httputil/timeoutreader.go

```diff
@@ -32,10 +32,14 @@ func (r *TimeoutReader) Read(p []byte) (int, error) {
 		n, err = r.reader.Read(p)
 		c <- 0
 	}()
+
+	timer := time.NewTimer(r.timeout)
+	defer timer.Stop()
+
 	select {
 	case <-c:
 		return n, err
-	case <-time.After(r.timeout):
+	case <-timer.C:
```

Comment:
This doesn't match the example in https://arangodb.com/2020/09/a-story-of-a-memory-leak-in-go-how-to-properly-use-time-after/
Seems like there can still be a message on the channel if the Stop fails (because the timer just fired while the code was reading from the other channel).
Can that message on timer.C be a memory leak? I can't tell for sure, but I'd suggest using exactly what is in the suggestion.